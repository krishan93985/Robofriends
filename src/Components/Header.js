import React, { memo } from 'react';
import Counter from './Counter';

//wrapping with memo does shallow props comparisons
//for within state comparison use useMemo (see docs)
//use shouldComponentUpdate for class based components
//to avoid shouldCom...Update use PureComponents in react
const Header = memo(() => {
    return (
        <div>
        <h1 className='f1'>Robofriends</h1>
        <Counter />
        </div>
    );
})

export default Header;