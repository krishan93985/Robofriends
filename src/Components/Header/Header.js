import React, { memo } from 'react';
import Counter from '../Counter/Counter';

//wrapping with memo does shallow props comparisons
//for within state comparison use useMemo (see docs)
//use shouldComponentUpdate for class based components
//to avoid shouldCom...Update use PureComponents in react
const Header = memo(() => {
    return (
        <div>
        <h1 style={{fontSize:"2.9rem", maxWidth:"100%"}}>Robofriends</h1>
        <Counter />
        </div>
    );
})

export default Header;