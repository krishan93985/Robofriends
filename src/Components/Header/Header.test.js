import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('is Header component correctly rendered', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
})