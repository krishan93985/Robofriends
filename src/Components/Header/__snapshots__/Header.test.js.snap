// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is Header component correctly rendered 1`] = `
<div>
  <h1
    style={
      Object {
        "fontSize": "2.9rem",
        "maxWidth": "100%",
      }
    }
  >
    Robofriends
  </h1>
  <Counter />
</div>
`;
